### ISSUE TITLE
Only send customer shipment (complete) emails when order isn't completed yet

### ISSUE THREAD

**Issue by strarsis** [vendidero/woocommerce-germanized#176]
When there is a bunch of existing, completed orders that still have shipments that aren't sent (postally) yet, 
and the stati of those shipments is changed to sent (postally), for each shipment an email would be sent to the customer.
If those completed orders are already some days/weeks/months ago, this would be confusing to the customer that probably already received and unboxed those shipments.

It would be nice if those shipment completed emails can only be sent if the order isn't completed yet –
or offer a toggle/checkbox to the backend user that prevents sending out those emails when a shipment status is changed to completed.

**Comment by dennisnissle** [comment]
Hi there,

why would you want to mark those shipments as shipped anyway? It doesn't make much sense. You could disable the email altogether within the email settings and (re) enable after the bulk-update.

**Comment by strarsis** [comment]
@dennisnissle: Follow up on this: I added a hook that, when a shipment status changes, checks whether the order status is already completed, before allowing the sending of a shipment-related email.

Additionally I configured Germanized to automatically set the status of shipments to completed when the order status is changed to completed. This helps a lot with forgetting to change the shipment status. The paradigm of orders and their shipments is not always immediately understood and orders usually have only one shipment, hence the order itself is treated equal to the shipment.
